packages/core/src/navigation-item/NavigationItem.tsx

```diff
@@ -75,34 +130,69 @@ export const NavigationItem = forwardRef<HTMLDivElement, NavigationItemProps>(
       "--saltNavigationItem-level": `${level}`,
     };
 
+    const isParent = parent || href === undefined;
+    const elementProps = {
+      className: clsx(
+        withBaseName("wrapper"),
+        {
+          [withBaseName("active")]: active || blurActive,
+          [withBaseName("blurActive")]: blurActive,
+          [withBaseName("rootItem")]: level === 0,
+        },
+        withBaseName(orientation)
+      ),
+      children: (
+        <>
+          <span className={withBaseName("label")}>{children}</span>
+          {isParent ? (
+            <ExpansionIcon expanded={expanded} orientation={orientation} />
+          ) : null}
+        </>
+      ),
+    };
+    let parentProps: Partial<ComponentPropsWithoutRef<"button">> = {};
+    let linkProps: Partial<ComponentPropsWithoutRef<"a">> = {};
+    if (isParent) {
+      const handleExpand = onExpand
+        ? (event: MouseEvent<HTMLButtonElement>) => {
+            event.stopPropagation();
+            onExpand(event);
+          }
+        : undefined;
+      parentProps = {
+        ...elementProps,
+        "aria-label": "expand",
+        "aria-expanded": expanded,
+        onClick: handleExpand,
+      };
+    } else {
+      linkProps = {
+        ...elementProps,
+        href,
+        "aria-label": "change page",
+        "aria-current": active ? "page" : undefined,
+      };
+    }
+
+    const defaultElementType = isParent ? "button" : "a";
+    const renderedElement = createElement(defaultElementType, {
+      active,
+      expanded,
+      parent: isParent,
+      level,
+      orientation,
+      render,
+      linkProps: !isParent ? linkProps : undefined,
+      parentProps: isParent ? parentProps : undefined,
```

Comment:
Just so my thoughts are noted:

- The downside of this approach is that you can't just put a HTML element in, you have to define logic how to handle non-native props